components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthApplicationMgtListener.java

```diff
@@ -463,11 +461,11 @@ private void removeEntriesFromCache(Set<String> consumerKeys) throws IdentityOAu
                 OAuthCache.getInstance().clearCacheEntry(new OAuthCacheKey(oauthKey));
             }
 
-            if (isNotEmpty(accessTokenDOSet)) {
+            if (isNotEmpty(accessTokenDOSet) && threadLocalForClaimConfigUpdates.get()) {
```

Comment:
@JeethJJ should we conditionally clear the cache against the tokens as well ? I thought it is only the authorization code.

Reply:
@DMHP will confirm if we are storing anything against the token that should not be cleared and provide an update here.

Reply:
@DMHP We are storing certain claims of a federated user against the token in the cache as well. Hence this validation is required.